src/utils/useOnOutsideClick.js

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useRef } from 'react';
+
+function useOnOutsideClick(callback) {
```

Comment:
Can we add proper type checking here? It is always being called as a function so I imagine at the least we should verify that its `Callable`